From 6e4fc03240f07dafa1e3cd76a3864c0ada386369 Mon Sep 17 00:00:00 2001
From: Huang Shijie <b32955@freescale.com>
Date: Tue, 18 Mar 2014 16:51:06 +0800
Subject: [PATCH 0369/1074] ENGR00303701-3 ARM: clk: imx6sx: use the 132MHz
 for the WEIM

We set the maximum clock frequency for the WEIM module.

Signed-off-by: Huang Shijie <b32955@freescale.com>
[Original patch taken from git://git.freescale.com/imx/linux-2.6-imx.git]
Signed-off-by: Biyao Zhai <biyao.zhai@windriver.com>
---
 arch/arm/mach-imx/clk-imx6sx.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-imx/clk-imx6sx.c b/arch/arm/mach-imx/clk-imx6sx.c
index f01e2af..659d572 100644
--- a/arch/arm/mach-imx/clk-imx6sx.c
+++ b/arch/arm/mach-imx/clk-imx6sx.c
@@ -437,6 +437,10 @@ static void __init imx6sx_clocks_init(struct device_node *ccm_node)
 		clk_prepare_enable(clks[IMX6SX_CLK_USBPHY2_GATE]);
 	}
 
+	/* Set the default 132MHz for EIM module */
+	clk_set_parent(clks[IMX6SX_CLK_EIM_SLOW_SEL], clks[IMX6SX_CLK_PLL2_PFD2]);
+	clk_set_rate(clks[IMX6SX_CLK_EIM_SLOW], 132000000);
+
 	/* set parent clock for LCDIF1 pixel clock */
 	clk_set_parent(clks[IMX6SX_CLK_LCDIF1_PRE_SEL], clks[IMX6SX_CLK_PLL5_VIDEO_DIV]);
 	clk_set_parent(clks[IMX6SX_CLK_LCDIF1_SEL], clks[IMX6SX_CLK_LCDIF1_PODF]);
-- 
1.7.5.4

